contrib/epee/src/readline_buffer.cpp

```diff
@@ -16,6 +17,8 @@ static std::string last_prompt;
 std::mutex line_mutex, sync_mutex, process_mutex;
 std::condition_variable have_line;
 
+std::vector<std::string> rdln::readline_buffer::completion_commands = {"exit"};
```

Comment:
The variables are private but accessed via public static functions that if invoked during the namespace initialization process will result in undefined behavior. Which is typically why refactoring adds a really sneaky bug - the static function has a hidden initialization dependency that is not guaranteed to trigger in all builds. As an example, `readline_buffers::add_completion(...)` cannot be invoked until `readline_buffers::completion_commands` is initialized. So if a future PR creates a `command_handler` object and calls `set_handler` in a namespace static, the result is undefined behavior. I point this out because I've seen this seemingly unlikely scenario happen before, and so I started using function local statics unless I had a good reason.